### ISSUE TITLE
Results on COCO

### ISSUE THREAD

**Issue by XuYin94** [gaviotas/AMN#2]
Thanks for your wonderful work in WSSS. Could you release the script on COCO and upload the corresponding checkpoint?

**Comment by gaviotas** [comment]
Hello, @YininKorea.

Sorry for the late reply.

Basically, our training code for COCO is highly borrowed from [RIB](https://github.com/jbeomlee93/RIB).

However, the current code for COCO is rather messy, I'm in the process of making it cleaner.

I will upload the training code for COCO and the corresponding checkpoint as soon as possible.

**Comment by XuYin94** [comment]
Many thanks for your reply. 

Another question is about your environment setting. 

When I executed your code on NVIDIA TITAN and 3090, I got a similar seed performance as the reported one in the paper, it is pretty good.

However, after training the IRN network with the generated r ir lables, I only got 71.74 mIoU (the reported one is 72.2) pseudo labels after refinement.

May I know the details of your experimental environment? was the IRN network trained on the same GPU as the AMN?

**Comment by gaviotas** [comment]
Yes, it is. The IRN network was trained using the same GPU as the AMN (using only a single gpu).

In addition, I uploaded the training code for COCO in [coco14](https://github.com/gaviotas/AMN/tree/coco14) branch.

Please check it and let me know if you have any further questions :)

Thanks.

**Comment by XuYin94** [comment]
Sorry, please check your [coco](https://github.com/gaviotas/AMN/tree/coco14/step) branch.

It seems that the codes for each step are still of VOC rather than for coco.

**Comment by gaviotas** [comment]
Sorry for the confusion.

There was an issue during the upload.

Currently, the training codes for coco14 have been uploaded.

Thanks.

**Comment by XuYin94** [comment]
May I know your training device, there is always about a 0.46 mIou gap between my reproduced seed result against your reported one (on Pascal Voc)?

Sorry for my wordy, since my current work is conducted on your study.

**Comment by gaviotas** [comment]
When training AMN and IRN networks (on PascalVOC 2012), the training device was an NVIDIA V100 GPU.

Can you show me the training script to get the reproduced result? Then, I will check it out and test it.

Thanks.

**Comment by YininKorea** [comment]
Here is my training script.

`# NEED TO SET
DATASET_ROOT=/root/Dataset/VOC/VOCdevkit/VOC2012
GPU=0

# 1. train a classification network and compute refined seed
# CUDA_VISIBLE_DEVICES=${GPU} python3 run_sample.py \
#     --voc12_root ${DATASET_ROOT} \
#     --num_workers 8 \
#     --train_cam_pass True
#     --cam_eval_thres 0.15 \
#     --make_cam_pass True \
#     --eval_cam_pass True \
#     --cam_to_ir_label_pass True \
#     --conf_fg_thres 0.30 \
#     --conf_bg_thres 0.05

# 2.1. train an attribute manipulation network
CUDA_VISIBLE_DEVICES=${GPU} python3 run_sample.py \
    --voc12_root ${DATASET_ROOT} \
    --num_workers 8 \
    --train_amn_pass True

# 2.2. generate activation maps and refined seed for boundary refinement
CUDA_VISIBLE_DEVICES=${GPU} python3 run_sample.py \
    --voc12_root ${DATASET_ROOT} \
    --num_workers 8 \
    --make_amn_cam_pass True \
    --eval_amn_cam_pass True \
    --amn_cam_to_ir_label_pass True \
    --conf_fg_thres 0.45 \
    --conf_bg_thres 0.15

# 3.1. train a boundary refinement network (IRN)
CUDA_VISIBLE_DEVICES=${GPU} python3 run_sample.py \
   --voc12_root ${DATASET_ROOT} \
    --num_workers 8 \
    --train_irn_pass True

# 3.2. generate the final pseudo-masks
CUDA_VISIBLE_DEVICES=${GPU} python3 run_sample.py \
    --voc12_root ${DATASET_ROOT} \
    --num_workers 8 \
    --make_sem_seg_pass True \
    --eval_sem_seg_pass True`

I could get similar results (72.03 mIoU) as you reported (72.2 mIoU) on the paper when using your provided IRN checkpoint to refine the seeds.

But when I train IRN on my device (TITAN), the final result is only 71.74 mIoU. 

Could you check it?

If it is verified to be caused by the device difference, I think that I need to report it in my write-up.

**Comment by XuYin94** [comment]
I made no change to your training script, and only uncomment lines 33-37.

I could get similar results (72.03 mIoU) as you reported (72.20) on the paper when using your provided IRN checkpoint.

But when I train IRN on my device (TITAN), the final result is only 71.74 mIoU. 

Could you check it?

If this gap is verified to be caused by the device difference, I think that I need to report it in my paper. 

Thank you !!

**Comment by gaviotas** [comment]
I uncommented lines 33-37 from the training script and tested on NVIDIA V100 GPU.

The result of training the IRN network twice is as follows: 71.8 mIoU and 72.0 mIoU.
There is a performance variance that comes from IRN refinement itself.

Please train the IRN network several times and check the performance.

If there is still a performance gap, other factors such as device differences are considered to have an impact.

Thanks.

**Comment by XuYin94** [comment]
Could you also share your configuration about MS coco segmentation training (on [deeplabv2](https://github.com/kazuto1011/deeplab-pytorch))?

**Comment by gaviotas** [comment]
Hi @YininKorea,

We specify the configuration for MS COCO 2014 in the supplementary material.
Please see Section A (Implementation Details - Segmentation network).

Thanks.

**Comment by KatameRonin** [comment]
> The result of training the IRN network twice is as follows: 71.8 mIoU and 72.0 mIoU. There is a performance variance that comes from IRN refinement itself.
> 
> Please train the IRN network several times and check the performance.

Hi, how does training the IRN multiple times make the results better? From the code, I don't see the IRN updating its own results on multiple iterations.
